pages/instructorAssessmentQuestionManualGrading/instructorAssessmentQuestionManualGrading.js

```diff
@@ -24,30 +20,52 @@ router.get('/', function (req, res, next) {
   sqlDb.queryOneRow(sql.select_question, params, (err, result) => {
     if (ERR(err, next)) return;
 
-    res.locals.question_id = result.rows[0].question_id;
-    res.locals.question_title = result.rows[0].question_title;
-    res.locals.max_points = result.rows[0].max_points;
+    Object.assign(res.locals, result.rows[0]);
 
-    var params = {
-      assessment_id: res.locals.assessment.id,
-      assessment_question_id: res.locals.assessment_question_id,
-    };
+    debug('render page');
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  });
+});
 
-    sqlDb.query(sql.select_instance_questions_manual_grading, params, function (err, result) {
-      if (ERR(err, next)) return;
-      res.locals.instance_questions_to_grade = result.rows.filter(isGradable);
-      res.locals.instance_questions_graded = result.rows.filter((iq) => !isGradable(iq));
-      debug('render page');
-      res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+router.get('/instances.json', function (req, res, next) {
+  debug('GET /instances.json');
+
+  var params = {
+    assessment_id: res.locals.assessment.id,
+    assessment_question_id: res.locals.assessment_question_id,
+  };
+
+  sqlDb.query(sql.select_instance_questions_manual_grading, params, function (err, result) {
+    if (ERR(err, next)) return;
+    result.rows.forEach((row) => {
+      // bootstrap-table does not like nulls as filter targets, set to 0 instead
+      Object.assign(row, { assigned_grader: row.assigned_grader || 0 });
     });
+    res.send({ instance_questions: result.rows });
   });
 });
 
 router.post('/', (req, res, next) => {
   if (!res.locals.authz_data.has_course_instance_permission_edit) return next();
-  if (req.body.__action === 'edit_question_points') {
+  if (req.body.__action === 'batch_action') {
+    const action_data = JSON.parse(req.body.batch_action_data) || {};
+    const instance_question_ids = Array.isArray(req.body.instance_question_id)
+      ? req.body.instance_question_id
+      : [req.body.instance_question_id];
+    const params = {
+      assessment_id: res.locals.assessment.id,
+      assessment_question_id: res.locals.assessment_question_id,
+      instance_question_ids,
+      requires_manual_grading: !!action_data?.requires_manual_grading,
+      assigned_grader: action_data?.assigned_grader,
```

Comment:
@nwalters512 @mwest1066 do we need any sort of validation here to ensure the assigned grader is a staff member in the course instance? I'm thinking that a malicious user with TA access in one course could potentially use this to get information (name, etc.) from a user_id, but maybe I'm overthinking.

Reply:
We should definitely validate. I'm not too concerned about malicious users (I could already use the "Staff" page to add someone to my course and see their name), but just as a matter of principle, we don't want to allow someone to assign to a user outside of their course instance.